### PR TITLE
CRAFTERCMS-2080

### PR DESCRIPTION
**Pull request by dejan-brkic**
Added calculate dependencies API

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/2080

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 1]
```diff
@@ -540,6 +542,80 @@ public void deleteSiteDependencies(String site) throws ServiceException {
...
+            pathsParams.clear();
+            pathsParams.addAll(targetPaths);
+        } while (!exitCondition);
+
+        return ancestors;
+    }
+
+    private List<Map<String, String>> calculatePublishingDependenciesForListFromDB(String site, Set<String> paths) {
+        Map<String, Object> params = new HashMap<String, Object>();
+        params.put("site", site);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java]: Shouldn't these strings be in a constant somewhere (this line and the lines below)?

File: src/main/resources/org/craftercms/studio/api/v1/dal/DependencyMapper.xml [hunk 2]
```diff
@@ -151,6 +151,42 @@
         </foreach>
     </select>
 
+    <select id="calculatePublishingDependenciesForList" parameterType="java.util.Map"
+            resultType="java.util.Map">
+        SELECT cd1.source_path, cd1.target_path
+        FROM (dependency cd1 INNER JOIN item_state co1 ON cd1.target_path = co1.path AND cd1.site = co1.site)
+        WHERE cd1.site = #{site}
+        AND cd1.source_path IN
+        <foreach item="path" index="index" collection="paths"
```
Comment (sumerjabri) [inline src/main/resources/org/craftercms/studio/api/v1/dal/DependencyMapper.xml:160]: More string literals, no?

File: src/main/webapp/default-site/scripts/rest/api/1/dependency/calculate-dependencies.post.groovy [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+// site
+    try {
+        if (StringUtils.isEmpty(site)) {
+            invalidParams = true
+            paramsList.add("site")
+        }
+    } catch (Exception exc) {
+        invalidParams = true
+        paramsList.add("site")
```
Comment (sumerjabri) [inline src/main/webapp/default-site/scripts/rest/api/1/dependency/calculate-dependencies.post.groovy]: More string literals in this file.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/resources/org/craftercms/studio/api/v1/dal/DependencyMapper.xml (hunk 2 above)
Comment (dejan-brkic) [reply src/main/resources/org/craftercms/studio/api/v1/dal/DependencyMapper.xml:160]: Per convo, not applicable on SQL mappers

**Pull request merged**